go-controller/pkg/annotation/node.go

```diff
@@ -0,0 +1,34 @@
+package annotation
+
+import (
+	"fmt"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+func EnsureNetworkID(watchFactory factory.NodeWatchFactory, nInfo util.NetInfo, networkID *int) error {
+	if networkID == nil {
+		return fmt.Errorf("nil network id pointer")
+	}
+	if *networkID != util.InvalidNetworkID {
+		return nil
+	}
+	nodes, err := watchFactory.GetNodes()
+	if err != nil {
+		return fmt.Errorf("failed to get nodes: %v", err)
+	}
+	for _, node := range nodes {
+		*networkID, err = util.ParseNetworkIDAnnotation(node, nInfo.GetNetworkName())
```

Comment:
why are we going through all the nodes? whta if I have 500 nodes?

Reply:
Sure, I will break as soon as we get the value.